app/api/utils.py

```diff
@@ -55,3 +57,44 @@ def cli_for_function(function, outfile, url):
         processed_df.to_csv(outfile, index=False)
     else:  # print to STDOUT
         print(processed_df.to_csv(index=False))
+
+def get_all_state_finals():
+    states_docs_urls = pd.read_csv("app/api/state_docs_urls.csv")
+    return states_docs_urls['Finals'].tolist()
+
+def get_all_priortize_entries():
+    entries, finals = [], []
+    for _, state_row in states_docs_urls.iterrows():
+        if state_row['Entry']:
+            entries.append(state_row['Entry'])
+        else:
+            finals.append(state_row['Final'])
+
+    return (entries, finals)
+
+def run_function_on_states(function, entries, finals, outputDir):
+    """
+    Call the function on every google sheets url for the specified states.
+    """
+    states_docs_urls = pd.read_csv("app/api/state_docs_urls.csv")
+
+    def process(state_row, column):
+        print("Running function %s on %s %s sheet..." % (function.__name__, state_row['State'], column))
+        url = csv_url_for_sheets_url(state_row[column])
+        df = pd.read_csv(url)
+        processed_df = function(df)
+        return processed_df
+
+    for _, state_row in states_docs_urls.iterrows():
+        if state_row['Entry'] and state_row['State'] in entries:
+            processed_df = process(state_row, 'Entry')
+            if not processed_df.empty:
```

Comment:
This assumes that the function will return a dataframe; some are just checks though rather than updates, and will return nothing. Could you update this script to expect that, and only try to write out the resulting dataframe if one is returned? I think it should be as simple as `if processed_df is not None and not processed_df.empty`

Reply:
Yup!